http/client.go

```diff
@@ -209,11 +209,13 @@ func (c *Client) roundTrip(r *Request) (resp *Response, err error) {
 	} else if h := r.getHeader("Host"); h != "" {
 		host = h
 	} else {
-		host = r.url.Host
+		host = r.url.Hostname()
```

Comment:
Host would be `localhost:port` and DNS lookup would take about 3-4 seconds before failing.